sphinx/ext/napoleon/docstring.py

```diff
@@ -872,6 +884,83 @@ def __init__(self, docstring: Union[str, List[str]], config: SphinxConfig = None
         self._directive_sections = ['.. index::']
         super().__init__(docstring, config, app, what, name, obj, options)
 
+    def _convert_type_spec(self, _type):
+        def recombine_set(tokens):
+            def combine_set(tokens):
+                in_set = False
+                set_items = []
+
+                for token in tokens:
```

Comment:
It seems this does not support a set that is not closed (ex. `{1, 2`). It is not good to suppress it silently.

Reply:
I think the error case should be warned. To suppress errors is not kind for users. I think all humans sometimes get mistakes. So we have to support such irregular cases.

Reply:
agreed, I'll do that

Reply:
my current approach for splitting the tokens can easily be broken by putting braces in the strings. I'll look into this a bit more